ftwhttp/connection.go

```diff
@@ -65,25 +64,16 @@ func (c *Connection) send(data []byte) (int, error) {
 
 }
 
-func (c *Connection) receive() ([]byte, error) {
+func (c *Connection) receive() (io.Reader, error) {
 	log.Trace().Msg("ftw/http: receiving data")
-	var err error
-	var buf []byte
 
 	// We assume the response body can be handled in memory without problems
 	// That's why we use io.ReadAll
-	if err = c.connection.SetReadDeadline(time.Now().Add(c.readTimeout)); err == nil {
-		buf, err = io.ReadAll(c.connection)
-	}
-
-	if neterr, ok := err.(net.Error); ok && !neterr.Timeout() {
-		log.Error().Msgf("ftw/http: %s\n", err.Error())
-	} else {
-		err = nil
+	if err := c.connection.SetReadDeadline(time.Now().Add(c.readTimeout)); err != nil {
+		return nil, err
 	}
-	log.Trace().Msgf("ftw/http: received data - %q", buf)
```

Comment:
Lost this, not sure of a good way to preserve it though. If anyone knows an alternative to this approach that still lets the response parsing finish without read timeout would be happy to change to it.

Reply:
Was able to use TeeReader

Reply:
Excellent! TIL.